src/runtime/TemporalObject.cpp

```diff
@@ -600,6 +600,201 @@ TemporalObject::DateTime TemporalObject::parseTemporalMonthDayString(ExecutionSt
+        }
+    } else {
+        auto parseResult = TemporalObject::parseTemporalZonedDateTimeString(state, value.asString()->toNonGCUTF8StringData());
+        calendar = TemporalCalendarObject::toTemporalCalendarWithISODefault(state, Value((parseResult.calendar)));
+
+        if (parseResult.tz->name->length() != 0) {
+            if (!TemporalTimeZoneObject::isValidTimeZoneName(parseResult.tz->name->toNonGCUTF8StringData())) {
+                ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "Invalid TimeZone name");
+            }
+
+            parseResult.tz->name = new ASCIIString(TemporalTimeZoneObject::canonicalizeTimeZoneName(parseResult.tz->name->toNonGCUTF8StringData()).c_str());
+            timeZone = TemporalTimeZoneObject::createTemporalTimeZone(state, parseResult.tz->name->toNonGCUTF8StringData());
+        }
+
+        if (parseResult.tz->z) {
+            offsetBehaviour = TemporalZonedDateTimeObject::EXACT;
+        } else if (parseResult.tz->offsetString->length() == 0) {
+            offsetBehaviour = TemporalZonedDateTimeObject::WALL;
+        }
+
+        matchBehaviour = TemporalZonedDateTimeObject::MINUTES;
+        result = {
+            { TemporalObject::YEAR_UNIT, parseResult.year },
+            { TemporalObject::MONTH_UNIT, parseResult.month },
+            { TemporalObject::DAY_UNIT, parseResult.day },
+            { TemporalObject::HOUR_UNIT, parseResult.hour },
+            { TemporalObject::MINUTE_UNIT, parseResult.minute },
+            { TemporalObject::SECOND_UNIT, parseResult.second },
+            { TemporalObject::MILLISECOND_UNIT, parseResult.millisecond },
+            { TemporalObject::MICROSECOND_UNIT, parseResult.microsecond },
+            { TemporalObject::NANOSECOND_UNIT, parseResult.nanosecond }
+        };
+    }
+
+    int64_t offsetNanoseconds = 0;
+
+    if (!timeZone.isUndefined()) {
+        if (offsetBehaviour == TemporalZonedDateTimeObject::OPTION) {
+            offsetNanoseconds = TemporalInstantObject::offsetStringToNanoseconds(state, offsetString.asString());
+        }
+
+        auto epochNanoseconds = TemporalZonedDateTimeObject::interpretISODateTimeOffset(state, result, offsetBehaviour, offsetNanoseconds, timeZone, Value(state.context()->staticStrings().lazycompatible().string()), Value(state.context()->staticStrings().reject.string()), matchBehaviour);
+        return TemporalZonedDateTimeObject::createTemporalZonedDateTime(state, *epochNanoseconds.asBigInt(), timeZone.asObject()->asTemporalTimeZoneObject(), calendar.asObject()->asTemporalCalendarObject());
+    }
+
+    return TemporalPlainDateObject::createTemporalDate(state, result[TemporalObject::YEAR_UNIT], result[TemporalObject::MONTH_UNIT], result[TemporalObject::DAY_UNIT], calendar);
+}
+
+TemporalObject::DateTimeUnits TemporalObject::largerOfTwoTemporalUnits(TemporalObject::DateTimeUnits u1, TemporalObject::DateTimeUnits u2)
+{
+    if (u1 > u2) {
+        return u2;
+    }
+
+    return u1;
+}
```

Comment:
This functin body is too simple to be a separate static function.
How about embedding the body rather than function call?